### PR TITLE
(PC-31935) feat(recommendation): more explicit error message

### PR DESCRIPTION
**Pull request by bruno-ebstein-pass-culture**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-31935

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [ ] Made sure my feature is working on web.
- [ ] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [x] Written **unit tests** native (and web when implementation is different) for my feature.

[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

### CODE REVIEW

File: src/libs/recommendation/useHomeRecommendedIdsQuery.ts [hunk 1]
```diff
@@ -36,15 +36,22 @@ export const useHomeRecommendedIdsQuery = (parameters: Parameters) => {
...
-        const statusCode = err instanceof ApiError ? err.statusCode : 'unknown'
-        const errorMessage = err instanceof Error ? err.message : JSON.stringify(err)
         const shouldApiErrorNotCaptured = Boolean(
           err instanceof ApiError && isAPIExceptionNotCaptured(err.statusCode)
         )
 
         if (!shouldApiErrorNotCaptured) {
+          const statusCode = err instanceof ApiError ? err.statusCode : 'unknown'
+          const title =
+            err instanceof ApiError
```
Comment (cedricls1) [inline src/libs/recommendation/useHomeRecommendedIdsQuery.ts:46]: Sonar n'a pas râlé ^^

Comment (cedricls1) [reply src/libs/recommendation/useHomeRecommendedIdsQuery.ts:46]: Edit : Il a bien râlé, une petite fonction pour cette partie ?

**Review by pierrecanthelou — APPROVED** [review]

File: src/libs/recommendation/useHomeRecommendedIdsQuery.ts (hunk 1 above)
Comment (bruno-ebstein-pass-culture) [reply src/libs/recommendation/useHomeRecommendedIdsQuery.ts]: bien vu !

File: src/libs/recommendation/useHomeRecommendedIdsQuery.ts [hunk 2]
```diff
@@ -51,42 +51,24 @@ export const useHomeRecommendedIdsQuery = (parameters: Parameters) => {
...
-    eventMonitoring.captureException(new Error(`Error ${title} with recommendation endpoint`), {
-      extra: {
-        playlistRequestBody: JSON.stringify(parameters.playlistRequestBody),
-        playlistRequestQuery: JSON.stringify(parameters.playlistRequestQuery),
-        statusCode,
-        errorMessage,
-      },
-    })
-    return
+  if (err instanceof ApiError) {
```
Comment (pierrecanthelou) [inline src/libs/recommendation/useHomeRecommendedIdsQuery.ts]: Risque-t-on d'avoir d'autres instanceof ?

**Review by cedricls1 — APPROVED** [review]

File: src/libs/recommendation/useHomeRecommendedIdsQuery.ts (hunk 2 above)
Comment (bruno-ebstein-pass-culture) [reply src/libs/recommendation/useHomeRecommendedIdsQuery.ts]: je ne sais pas, je ne pense pas

**Pull request merged**